StudyProgramme Refactor: Move external StudyProgramme Code out of Certificate

fhelfer opened the pull request:
Hi @klees,

With this PR we would like to move consumer specific classes from components/ILIAS/Certificate to the respective component, like it has been already done for CmiXapi and LTIConsumer.

All related unit tests were moved as well, they are added to the unit tests suites of the particular component.

There are still parts of tighter coupling to some object type strings in components/ILIAS/Certificate (e.g. to register a consumer), but this will hopefully be removed in future when the component revision has been finalized and all components are revised accordingly (funding might be necessary of course, for the certificate component as well as for the certificate providing components).

Feel free to organize the structure or rename the classes according to your component specific rules/patterns/guidelines.

Best, @fhelfer

klees reviewed (changes requested):
Hi @fhelfer,

I appreciate the intended move. Its better if other components know the Certificate component instead of the Certificate component knowing all other components.

I find it a little hard to really understand what the exact change is here. This seems to contain a lot of unrelated changes (e.g. introduction of `use` statements, exchange of `"` to `'`) that seem to reflect personal preferences, possibly introduced by some automatism of an IDE.

Could you please strip this down to the changes that are actually required to make the intended change?

Kind regards!

mjansenDatabay commented:
@fhelfer Please be aware that the class name of the `ilCertificatePlaceholderValues`-derivative is stored in the database (`il_cert_cron_queue`). Simply renaming this class (or intrdoducing namespaces) will cause errors.

fhelfer committed:
Move external StudyProgramme Code out of Certificate

StudyProgramme Certificate Update Step replacing adapterclass

Rename StudyProgramme UpdateSteps for Certificate

add StudyProgramme UpdateAgent to contribute step

fix typo in update name

fhelfer commented:
Hi @klees,

removed all changes except introduction of a new namespace for Certificates in StudyProgrammes. Please let me know if you also want this to be removed
In addition, based on the hint @mjansenDatabay gave, I added a new UpdateStep changing the adapter class in il_cert_con_queue to according to the new namespace for PlacholderVlaue class.

Best @fhelfer

klees commented:
Thx @fhelfer! Still plausible =)
